src/SprykerEco/Zed/Payone/Business/Payment/Reader/PayoneSecurityInvoiceReader.php

```diff
@@ -0,0 +1,128 @@
+use SprykerEco\Zed\Payone\Business\Payment\DataMapper\StandartParameterMapperInterface;
+use SprykerEco\Zed\Payone\Business\Payment\PaymentMapperReaderInterface;
+use SprykerEco\Zed\Payone\Persistence\PayoneQueryContainerInterface;
+
+class PayoneSecurityInvoiceReader implements PayoneSecurityInvoiceReaderInterface
+{
+    public const ERROR_ACCESS_DENIED_MESSAGE = 'Access denied';
+
+    /**
+     * @var \SprykerEco\Zed\Payone\Business\Api\Adapter\AdapterInterface
+     */
+    protected $executionAdapter;
+
+    /**
+     * @var \SprykerEco\Zed\Payone\Persistence\PayoneQueryContainerInterface
+     */
+    protected $queryContainer;
+
+    /**
+     * @var \Generated\Shared\Transfer\PayoneStandardParameterTransfer
+     */
+    protected $standardParameter;
+
+    /**
+     * @var \SprykerEco\Zed\Payone\Business\Payment\DataMapper\StandartParameterMapperInterface
+     */
+    protected $standartParameterMapper;
+
+    /**
+     * @var \SprykerEco\Zed\Payone\Business\Payment\PaymentMapperReaderInterface
+     */
+    protected $paymentMapperReader;
+
+    /**
+     * @param \SprykerEco\Zed\Payone\Business\Api\Adapter\AdapterInterface $executionAdapter
+     * @param \SprykerEco\Zed\Payone\Persistence\PayoneQueryContainerInterface $queryContainer
+     * @param \Generated\Shared\Transfer\PayoneStandardParameterTransfer $standardParameter
+     * @param \SprykerEco\Zed\Payone\Business\Payment\DataMapper\StandartParameterMapperInterface $standartParameterMapper
+     * @param \SprykerEco\Zed\Payone\Business\Payment\PaymentMapperReaderInterface $paymentMapperReader
+     */
+    public function __construct(
+        AdapterInterface $executionAdapter,
+        PayoneQueryContainerInterface $queryContainer,
+        PayoneStandardParameterTransfer $standardParameter,
+        StandartParameterMapperInterface $standartParameterMapper,
+        PaymentMapperReaderInterface $paymentMapperReader
+    ) {
+        $this->executionAdapter = $executionAdapter;
+        $this->queryContainer = $queryContainer;
+        $this->standardParameter = $standardParameter;
+        $this->standartParameterMapper = $standartParameterMapper;
+        $this->paymentMapperReader = $paymentMapperReader;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\PayoneGetSecurityInvoiceTransfer $getSecurityInvoiceTransfer
+     *
+     * @return \Generated\Shared\Transfer\PayoneGetSecurityInvoiceTransfer
+     */
+    public function getSecurityInvoice(PayoneGetSecurityInvoiceTransfer $getSecurityInvoiceTransfer): PayoneGetSecurityInvoiceTransfer
+    {
+        $responseContainer = new GetSecurityInvoiceResponseContainer();
+        $paymentEntity = $this->findPaymentByInvoiceTitleAndCustomerId(
+            $getSecurityInvoiceTransfer->getReference(),
+            $getSecurityInvoiceTransfer->getCustomerId()
+        );
+
+        if (!$paymentEntity) {
+            $this->setAccessDeniedError($responseContainer);
```

Comment:
* Set the errors to a `$getSecurityInvoiceTransfer`.

Reply:
Thanks